### PR TITLE
Optimize filtering of RelayHub events

### LINKED ISSUE #14: Review events processing on relay client
According to https://github.com/tabookey/tabookey-gasless/issues/119, events processing for relayer discovery can be improved.

### PR DESCRIPTION
**Pull request by frangio**
Fixes #14.

Previously we were retrieving all of the events instead of asking the node for only `RelayAdded` and `RelayRemoved` events.

### CODE REVIEW

File: test/GSNProvider.test.js [hunk 1]
```diff
@@ -518,6 +519,59 @@ describe('GSNProvider', function () {
...
+
+  context('detecting relays added', function () {
+    before('getting relayHub', async function () {
+      this.relayHub = await getRelayHub(this.web3);
+    });
+
+    beforeEach('creating provider', async function () {
+      this.gsnProvider = new GSNProvider(PROVIDER_URL, HARDCODED_RELAYER_OPTS);
+      this.greeter.setProvider(this.gsnProvider);
+      await this.greeter.methods.greet("Hello").send({ from: this.sender, useGSN: true });
```
Comment (spalladino) [inline test/GSNProvider.test.js]: Why is this line needed?

File: test/GSNProvider.test.js [hunk 2]
```diff
@@ -518,6 +519,59 @@ describe('GSNProvider', function () {
...
+      expect(relays).to.have.lengthOf(1);
+    });
+
+    context('with a second relay added', async function () {
+      beforeEach('adding second relay', async function () {
+        await this.relayHub.methods.stake(this.secondRelay, 60*60*24*10 /* 10 days */)
+          .send({
+            value: Web3.utils.toWei('1', 'ether'),
+            from: this.deployer,
+            useGSN: false,
```
Comment (spalladino) [inline test/GSNProvider.test.js]: `this.relayHub` should be connected to a vanilla provider, so no need for the useGSN flag here

Comment (frangio) [reply test/GSNProvider.test.js]: Are you okay with leaving it just for explicitness?

File: test/GSNProvider.test.js (hunk 1 above)
Comment (frangio) [reply test/GSNProvider.test.js]: It's needed because `RelayClient`

Comment (spalladino) [reply test/GSNProvider.test.js]: Good point. Let's leave it like this for now.

File: test/GSNProvider.test.js (hunk 2 above)
Comment (spalladino) [reply test/GSNProvider.test.js]: Definitely
